pkg/providers/vsphere/vmprovider_resourcepolicy.go

```diff
@@ -190,6 +191,24 @@ func (vs *vSphereVMProvider) createClusterModules(
 		}
 	}
 
+	desiredGroups := sets.New(resourcePolicy.Spec.ClusterModuleGroups...)
```

Comment:
This isn't the right place for this - this function is handle the cluster modules within just a CCR so this is duplicate work.

One existing issue with this code that I was planning to fix under this work is that when a zone is removed from a NS the cluster modules for that CCR should also be removed. Right now that won't happen until the VMSetRP is deleted.

Reply:
Ah got it, I agree.

Refactored the code to handle compute cluster resources in its separate loop inside the reconcile function itself.